### PR TITLE
Use docker-compose for running stress tests

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Changes the linux pipeline so that client and server and run in separate containers. Also makes changes to the stress suite to accomodate hosting the server-side in a standalone container.

### CODE REVIEW

File: src/System.Net.Http/tests/StressTests/HttpStress/Program.cs [hunk 1]
```diff
@@ -124,7 +124,7 @@ private static async Task<ExitCode> Run(Configuration config)
             return ExitCode.CliError;
         }
 
-        if (!config.ServerUri.Scheme.StartsWith("http"))
+        if (!config.ServerUri.StartsWith("http"))
```
Comment (stephentoub) [inline src/System.Net.Http/tests/StressTests/HttpStress/Program.cs]: Just curious why this changed?

File: src/System.Net.Http/tests/StressTests/HttpStress/StressClient.cs [hunk 2]
```diff
@@ -80,7 +82,8 @@ public void PrintFinalReport()
 
         private async Task StartCore()
         {
-            if (_config.ServerUri.Scheme == "http")
+
```
Comment (stephentoub) [inline src/System.Net.Http/tests/StressTests/HttpStress/StressClient.cs]:
```suggestion
```

File: src/System.Net.Http/tests/StressTests/HttpStress/StressClient.cs [hunk 3]
```diff
@@ -147,21 +172,24 @@ int Combine(int h1, int h2)
...
             }
-
-            // Spin up a thread dedicated to outputting stats for each defined interval
-            new Thread(() =>
+            
+            async Task SendTestRequestToServer(int maxRetries)
             {
-                while (!_cts.IsCancellationRequested)
+                using HttpClient client = CreateHttpClient();
+                for(int remainingRetries = maxRetries; ; remainingRetries--)
```
Comment (stephentoub) [inline src/System.Net.Http/tests/StressTests/HttpStress/StressClient.cs]:
```suggestion
                for (int remainingRetries = maxRetries; ; remainingRetries--)
```

File: src/System.Net.Http/tests/StressTests/HttpStress/StressServer.cs [hunk 4]
```diff
@@ -309,6 +322,28 @@ private void SetUpJustInTimeLogging()
...
+        private static (string scheme, string hostname, int port) ParseServerUri(string serverUri)
+        {
+            try
+            {
+                var uri = new Uri(serverUri);
+                return (uri.Scheme, uri.Host, uri.Port);
+            } 
+            catch (UriFormatException)
+            {
+                // Poor man's uri parser: used to parse values valid in Kestrel
```
Comment (stephentoub) [inline src/System.Net.Http/tests/StressTests/HttpStress/StressServer.cs]:
```suggestion
                // Simple uri parser: used to parse values valid in Kestrel
```

File: src/System.Net.Http/tests/StressTests/HttpStress/Program.cs (hunk 1 above)
Comment (eiriktsarpalis) [reply src/System.Net.Http/tests/StressTests/HttpStress/Program.cs]: `ServerUri` is no longer of type `System.Uri`, to support values such as `http://+:5050`. This is required for exposing the server port in containers.

**Pull request merged**